elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,76 @@
+= [WFCORE-4394] Enhanced Audit Logging - RFC support and reliability/speed customization
+:author:            Justin Michael Cook
+:email:             jucook@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+Currently, Audit Logging in Elytron does not have the same capabilities as Legacy Security. This task is to enhance Elytron Audit Logging to have support for RFC5424/RFC3164 and add the additional ability to customize reliability vs speed.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4394[WFCORE-4394]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-699[EAP7-699]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kašík]
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Support for https://tools.ietf.org/html/rfc5424[RFC 5424] should be enhanced, according to the following parameter,
+which should be added to ```syslog-audit-log``` in the Elytron subsystem
+** An optional ```syslog-format``` attribute should be added which supports a value of ```RFC5424```
+** This attribute will also have ```RFC5424``` as the default value
+** A Rfc5424SyslogEvent should be added as an audit event, which will trigger when the Elytron Subsystem attempts to log to a syslog server with the format ```RFC5424```
+
+* Support for https://tools.ietf.org/html/rfc3164[RFC 3164] should be added to ```syslog-audit-log``` in the Elytron subsystem
+** The ```syslog-format``` attribute described in the RFC 5424 section above should also support a value of ```RFC3164```
+** A Rfc3164SyslogEvent should be added as an audit event, which will trigger when the Elytron Subsystem attempts to log to a syslog server with the format ```RFC3164```
+
+* The ability to configure for less reliability for increased speed should be added
+** This should be done by configuring the number of reconnect attempts to a syslog-server
+** This should be available as an attribute ```reconnect-attempts``` under ```syslog-audit-log``` in the Elytron subsystem
+** The possible values for this parameter would be: -1 - signifying indefinite reconnect attempts,
```

Comment:
@justinmcook From our discussion I understood that currently there is only on reconnect attempt made (and error is ignored). From my point of view we should keep backward compatibility and keep this as is by default. Can you please find out if current behavior is intended (if it's not a bug)?

cc @fjuma